packages/material-ui/src/Modal/TrapFocus.js

```diff
@@ -1,25 +1,32 @@
 /* eslint-disable consistent-return, jsx-a11y/no-noninteractive-tabindex */
 
 import React from 'react';
+import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
 import warning from 'warning';
-import RootRef from '../RootRef';
 import ownerDocument from '../utils/ownerDocument';
+import { useForkRef } from '../utils/reactHelpers';
 
 function TrapFocus(props) {
   const {
+    children,
     disableAutoFocus,
     disableEnforceFocus,
     disableRestoreFocus,
     getDoc,
     isEnabled,
     open,
   } = props;
-  const rootRef = React.useRef();
   const ignoreNextEnforceFocus = React.useRef();
   const sentinelStart = React.useRef();
   const sentinelEnd = React.useRef();
   const lastFocus = React.useRef();
+  const rootRef = React.useRef();
+  // can be removed once we drop support for non ref forwarding class components
+  const handleOwnRef = React.useCallback(ref => {
+    rootRef.current = ReactDOM.findDOMNode(ref);
```

Comment:
I believe we can add the `// StrictMode ready` comment. If @eps1lon accepts it.

Reply:
Yes. Actually every findDOMNode that get's an instance from a ref should be strict-mode.